app/scripts/controllers/rif/rns/resolver/index.js

```diff
@@ -128,25 +128,47 @@ export default class RnsResolver extends RnsJsDelegate {
     });
   }
 
-  // this.rifConfig.rns.contracts.multiChainResolver This is the path to a multicahin address in the config
-  // TODO: Rodrigo
-  getNetoworksForResolvers (address) {
+  /**
+   * Returns an array of network addresses for a given domain
+   * @param domainNameResolver DomainName with the .rsk extension
+   * @returns {Promise<unknown>}
+   */
+  getNetworksForResolvers (domainNameResolver) {
     return new Promise((resolve, reject) => {
+      // TODO: Rodrigo
       // This event is different, it deppends if we're on the chain of RSK, or not, to develop purposes i'll use only ChainAddrChanged, it need to differentiate it from one to another
-      console.debug('INICIO');
-
-      const myEvent = this.multiChainresolverContractInstance.ChainAddrChanged({node: namehash.hash('testing.rsk')}, {fromBlock: 0, toBlock: 'latest'});
+      const myEvent = this.multiChainresolverContractInstance.ChainAddrChanged({node: namehash.hash(domainNameResolver)}, {fromBlock: 0, toBlock: 'latest'});
       myEvent.get(function(error, result){
-        console.debug('RESULT ===================', result);
-        console.debug('ERROR ===================', error);
+        if (error) {
+          console.debug('Error when trying to get netoworks for resolver', error);
+          reject(error);
+        }
+        const arrChains = [];
+        result.forEach(network => {
+          const networkToPush = new ResolverNetwork(network.args.chain, network.args.addr)
+          const index = arrChains.findIndex((e) => e.chain === networkToPush.chain);
```

Comment:
I dont think this is ok. 

If we have, for example, these events: 

0, 0x1
2, 0x1
3, 0x2
3, 0x4

This will search for the second occurrence of the chain id 3 (3, 0x4), and overrides the previous (3, 0x2) entry. And thats not ok, because those are two different addresses. 

We need to get the latest event for the same address and the same chainId.

Reply:
This works fine, because it's filtering by domain name in this line: 
`chainAddrChangedEvent = this.multiChainresolverContractInstance.ChainAddrChanged({node: namehash.hash(domainNameResolver)}, {fromBlock: 0, toBlock: 'latest'});`
So it will be looking for events for a particular domainName. Iterating in the forEach, its replacing the older objects in the array to return.

Reply:
Thats right!